express-notes-api/post.js

```diff
@@ -0,0 +1,22 @@
+module.exports = function post(app, data, path, fs, write) {
+  app.post('/api/notes', (req, res) => {
+    const body = req.body;
+    if (body.content === undefined) {
+      const notFound = { error: 'content is a required field' };
+      res.status(400).json(notFound);
+      return;
+    }
+    const id = data.nextId;
+    data.notes[id] = req.body;
+    data.notes[id].id = id;
+    data.nextId++;
+    write(path, data);
+    if (!fs.existsSync(path)) {
```

Comment:
You are checking if the file exists _after_ you've written to it! Check it before and don't do the write if it doesn't exist.